packages/typescript/src/generated.ts

```diff
@@ -19,6 +19,7 @@ import {
 export enum Type {
   Load = "mx/load",
   Ping = "mx/ping",
+  Navigation = "mx/navigation",
```

Comment:
I'm late to the party, but `navigation` to me seems like this could be fired on *any* navigation (backwards, forwards, to a completely new page, etc) at all but we are focused on using this exclusively as a "go back" initiative. Should we further namespace this as:

`mx/navigation/:direction`

or something similar? I don't really anticipate, or understand, how we would work with forwards/redirect with what we currently have, and don't want people to think that this navigation event is more than what it is.

Reply:
I think that no matter what or how we do it, we want to keep this as an "internal" message that we keep customers from using. Maybe we make the warning sound more severe or even hide this field from the docs altogether, but I'm not sure splitting the events into a back/forward gives us much. If we want direction info, `mx/navigation` could have a `did_go_forward` field in the future and do the same thing. I think we just keep this message out of our documentation and keep customers from using it by making it hard to find.

Reply:
I would be 100% on board with this not being documented, and maybe add a field or something to the event itself that has some sort of message that makes it obvious to anyone tinkering that they should *not* be using it for anything.

Reply:
Maybe we could add a `private` field to it, and just add a note in the docs that some post messages are private, not meant to be used by clients, and can change at *any* time without notice?

Reply:
RE the above, I'm gonna make an update to this project so support optional fields. Once that's in place, I don't think an update to exclude certain fields from the docs would be too hard. How about we let this move up as-is and then follow it up with the update that removes this post message from the docs? I can work on the first part (the optional field bit) that enables this, then @inphomercial can take the work that excludes fields from the docs. Thoughts @sambev @inphomercial ?

Reply:
https://github.com/mxenabled/widget-post-message-definitions/pull/15 adds support for a "documented" flag which when set to "false" will exclude the post message from the docs. We can flag these fields as `documented: false` once that PR is released.